File: src/backend/auth/restaurant_management_system_spring_boot/src/main/java/com/example/demo/entities/Category.java
package com.example.demo.entities;

import java.util.Set;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Table(name="categories")
public class Category {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	int c_id;

	String c_name;

	@OneToMany(mappedBy = "c_id", cascade = CascadeType.ALL)
	@JsonIgnore
	Set<Subcategory> subcategories;

}

File: src/backend/dotNet/restaurant_management_system/waiter_srv/Models/Category.cs
using System;
using System.Collections.Generic;

namespace waiter_srv.Models;

public partial class Category
{
    public int CId { get; set; }

    public string? CName { get; set; }

    public virtual ICollection<Subcategory> Subcategories { get; set; } = new List<Subcategory>();
}
